lib/Parse/ParseExpr.cpp

```diff
@@ -422,6 +422,20 @@ ParserResult<Expr> Parser::parseExprSequenceElement(Diag<> message,
     return sub;
   }
 
+  if (Tok.isContextualKeyword("copy") &&
+      peekToken().isAny(tok::identifier, tok::kw_self) &&
+      !peekToken().isAtStartOfLine()) {
+    Tok.setKind(tok::contextual_keyword);
+
+    SourceLoc copyLoc = consumeToken();
+    ParserResult<Expr> sub =
+        parseExprSequenceElement(diag::expected_expr_after_copy, isExprBasic);
+    if (!sub.hasCodeCompletion() && !sub.isNull()) {
+      sub = makeParserResult(new (Context) CopyExpr(copyLoc, sub.get()));
+    }
```

Comment:
Why can’t we form a `CopyExpr` result if `sub` contains the code completion token?

Reply:
I don't know. I think I was just copying what we already had. Should I allow for that?

Reply:
I don’t see a reason why we shouldn’t.

Reply:
Thanks again for your help @ahoppen!